python/tvm/autotvm/graph_tuner/dynamic_programming_stage.py

```diff
@@ -0,0 +1,340 @@
+
+            if not input_node_time_counted:
+                self._global_counted_nodes_set.add(input_idx)
+            self._full_states = self._full_states.reshape(full_states_shape)
+
+            # If out degree of input node is 1, we can remove the dimension of input node,
+            # since the states of input node will not be needed any more. Otherwise, input
+            # node should become a dependency.
+            if len(self._global_out_nodes_dict[input_idx]) == 1:
+                self._states = np.amin(self._full_states, axis=1)
+                self._dep = list(input_dep)
+            else:
+                self._states = self._full_states
+                self._dep = [input_idx,] + input_dep
+
+        # Update global dependency dictionary.
+        # This is to monitor the dependency states to decide
+        # when a dependency can be eliminated, so that total
+        # number of states can be largely reduced.
+        for dep_idx in self._dep:
+            self._global_dep_dict[dep_idx].remove(self._idx)
+            for child in self._global_out_nodes_dict[self._idx]:
+                self._global_dep_dict[dep_idx].add(child)
+        if len(self._global_out_nodes_dict[self._idx]) > 1:
+            self._global_dep_dict[self._idx] = set()
+            for child in self._global_out_nodes_dict[self._idx]:
+                self._global_dep_dict[self._idx].add(child)
+
+    def _create_multi_inputs_states(self):
+        """State creation routine for multi_input operator"""
+        full_input_node_list = list(self._global_in_nodes_dict[self._idx])
+        input_index_list = []
+        # Remove input and parameter nodes
+        for input_idx in full_input_node_list:
+            if not is_input_node(self._global_node_list[input_idx],
+                                 self._global_input_names):
+                input_index_list.append(input_idx)
+
+        # Generate new states
+        states_list, aligned_node_list = DPStage.align_states(input_index_list,
+                                                              self._global_stage_dict,
+                                                              self._global_node_list)
+        aligned_shape = states_list[0][3].shape
+        self._full_states = np.zeros(aligned_shape).astype("float32").flatten()
+        self._full_states_idx = list(aligned_node_list)
+        num_states = self._full_states.shape[0]
+        node_time_counted = [item[0] in self._global_counted_nodes_set for item in states_list]
+        target_node_idx = states_list[0][0]
+        target_states = states_list[0][3].flatten()
+        target_multiplier = states_list[0][2]
+        target_major_axis = states_list[0][1]
+        src_states_list = [states_list[i][3].flatten() for i in range(1, len(states_list))]
+
+        for i in range(num_states):
+            target_sch_idx = (i % (target_multiplier *
+                                   aligned_shape[target_major_axis])) // target_multiplier
```

Comment:
trying to understand why

Reply:
Added docstring for this method to explain more details.